server/server.js

```diff
@@ -0,0 +1,13 @@
+var express = require('express');
+var cors = require('cors');
+var path = require('path');
+var bodyParser = require('body-parser');
+var app = express();
```

Comment:
Why do you use var? I'm only a beginner so I don't mean that in a condescending way at all, and more of a teaching moment for me as well! I was taught that let is the new var, and so I'd love to hear why var is still good to use!

Reply:
Hey Logan!  I appreciate the question!  
So while it is pretty safe in most cases to just replace VAR with LET in your vocabulary, it is important to understand that they are fundamentally very different.  The scope of a variable declared with VAR is its _**execution context**_.  Var declarations, wherever they occur, are processed before any code is executed. This is called hoisting.   [You can check out MDN's article on Var declarations here!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var)
It provides some really great examples of what Var hoisting does, but a quick example is this:
`bla = 2;
var bla;

// ...is implicitly understood as:

var bla;
bla = 2;`

Variables declared with LET are scoped to their 'block' and are not hoisted.  One really great example of where you will likely see this create issues is this:
`function logStuff(arr ){
  for(var i = 0; i < arr.length; i++){
  console.log(i)
}
console.log(i);
}`
If given an array with length of 5, this function will log:
`0
1
2
3
4
4`
Because var i =0 allows for accessing i outside of the block scope of the for loop.  

Conversely, 
`function logStuff(arr ){
  for(let i = 0; i < arr.length; i++){
  console.log(i)
}
console.log(i);
}`
If given an array with length of 5 logs: 
`0
1
2
3
4`
And then throw an error as the console.log(i) written outside of the block scope of the forLoop does not have access to any variable named i.  Some modern IDE's will log that last console.log(i) as undefined, but your browser is very likely to throw an error and your IDE should at least inform you that it is going to error if it doesn't immediately throw one.

Reply:
![Screen Shot 2020-04-04 at 4 49 19 PM](https://user-images.githubusercontent.com/53983066/78462066-426e8c80-7694-11ea-83b3-8437dd92d256.png)

Reply:
Kytra, I really appreciate the in-depth response...! You could make a blog post of this one.